src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,206 @@
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level; _⊔_)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties          as Properties
+import Function.Definitions
+import Relation.Binary.Reasoning.Setoid   as Reasoning
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Agda.Builtin.Sigma
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+  using (Σ; _,_; ∃; Σ-syntax; ∃-syntax; _×_)
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+open Module modA
+  using () renaming
+  ( Carrierᴹ       to A
+  ; _+ᴹ_           to _+ᴬ_
+  ; _*ₗ_           to _·ᴬ_
+  ; -ᴹ_            to -ᴬ_
+  ; _≈ᴹ_           to _≈ᴬ_
+  ; 0ᴹ             to 0ᴬ
+  ; +ᴹ-comm        to +ᴬ-comm
+  ; +ᴹ-congˡ       to +ᴬ-congˡ
+  ; +ᴹ-congʳ       to +ᴬ-congʳ
+  ; *ₗ-zeroˡ       to ·ᴬ-zeroˡ
+  ; -ᴹ‿cong        to -ᴬ‿cong
+  ; -ᴹ‿inverseʳ    to -ᴬ‿inverseʳ
+  ; -ᴹ‿involutive  to -ᴬ‿involutive
+  ; uniqueʳ‿-ᴹ     to uniqueʳ‿-ᴬ
+  ; ≈ᴹ-setoid      to ≈ᴬ-setoid
+  ; ≈ᴹ-sym         to symᴬ
+  ; leftSemimodule to leftSemimoduleᴬ
+  )
+open Properties leftSemimoduleᴬ
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴬ
+  ; non-zeroˡ to non-zeroˡᴬ
+  )
+open Module modB
+  using () renaming
+  ( Carrierᴹ       to B
+  ; _+ᴹ_           to _+ᴮ_
+  ; _*ₗ_           to _·ᴮ_
+  ; -ᴹ_            to -ᴮ_
+  ; _≈ᴹ_           to _≈ᴮ_
+  ; 0ᴹ             to 0ᴮ
+  ; +ᴹ-comm        to +ᴮ-comm
+  ; +ᴹ-congˡ       to +ᴮ-congˡ
+  ; +ᴹ-congʳ       to +ᴮ-congʳ
+  ; *ₗ-zeroˡ       to ·ᴮ-zeroˡ
+  ; -ᴹ‿cong        to -ᴮ‿cong
+  ; -ᴹ‿inverseʳ    to -ᴮ‿inverseʳ
+  ; -ᴹ‿involutive  to -ᴮ‿involutive
+  ; uniqueʳ‿-ᴹ     to uniqueʳ‿-ᴮ
+  ; ≈ᴹ-setoid      to ≈ᴮ-setoid
+  ; ≈ᴹ-sym         to symᴮ
+  ; leftSemimodule to leftSemimoduleᴮ
+  )
+open Properties leftSemimoduleᴮ
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴮ
+  ; non-zeroˡ to non-zeroˡᴮ
+  )
+open CommutativeRing ring
+  using (_≈_; _*_; _+_) renaming
+  ( Carrier to S
+  ; 0#      to 𝟘
+  ; 1#      to 𝟙
+  )
+open module Reasoningᴬ = Reasoning ≈ᴬ-setoid
+  using () renaming
+  ( begin_ to beginᴬ_
+  ; _∎     to _∎ᴬ
+  )
+infixr 2 step-≈ᴬ
+step-≈ᴬ = Reasoningᴬ.step-≈
+syntax step-≈ᴬ x y≈z x≈y = x ≈ᴬ⟨ x≈y ⟩ y≈z
+open module Reasoningᴮ = Reasoning ≈ᴮ-setoid
+open Function.Definitions _≈ᴬ_ _≈ᴮ_
+
+_≉ᴬ_ : A → A → Set ℓm
+x ≉ᴬ y = ¬ (x ≈ᴬ y)
+
+_≉ᴮ_ : B → B → Set ℓm
+x ≉ᴮ y = ¬ (x ≈ᴮ y)
+
+module _
+  {⟦_⟧ : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism ⟦_⟧)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  -- f(x) ≈ 0 iff x ≈ 0, for linear non-trivial f
+  f𝟘≈𝟘 : {x : A} → ⟦ 0ᴬ ⟧ ≈ᴮ 0ᴮ
```

Comment:
This is kind of a weird name, as there is no `f` in the rhs.  You should be consistent about what you name the homomorphism. I have no strong preference between `⟦_⟧` and `f`, but strong preference that you make a choice.

I have a weak preference for `f`. `⟦_⟧` is "semantics brackets", and this isn't quite semantics.

Reply:
Also, this is a really weird property! You have an implicit `x` in the assumptions that is not used in the conclusion. It's never ever going to be inferred, so it will always need to be given.

I think you can use `1` instead of `x`. Your proof just needs that `A` is inhabited to go through, and `1` gives you that. You could even use `0` !  But `x` is bad.

Reply:
D'oh! Yes, I'd intended to change all `f`s to `⟦_⟧`s and missed this one. :(

Oh, _very_ interesting that you actually prefer `f`. I do, as well, as I'd much prefer that `⟦_⟧` only ever meant, "the meaning of." I guess I misread an intended convention when perusing the other code in the standard library.
I'll revert to `f`.

I'll also get rid of the unnecessary `x`.

Reply:
I believe that I've addressed everything here.

**These changes included in latest push and ready for review.**